tests/unit/dataset/loader/test_mooncake_payload_mode.py

```
@@ -8,31 +8,62 @@
+    t = MooncakeTrace(
+        text_input="Hello",
+        output_length=3,
+        output_token_ids=[10, 11, 12],
+        request_id="request-a",
+    )
+    assert t.output_token_ids == [10, 11, 12]
+    assert t.request_id == "request-a"
+
+
+def test_mooncake_trace_rejects_replay_length_mismatch() -> None:
+    with pytest.raises(ValidationError, match="output_length.*len"):
```

Ruff (0.15.18)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)